main.rb

```diff
@@ -0,0 +1,64 @@
+require 'open3'
+
+def get_env_variable(key)
+    return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+ 
+def env_has_key(key)
```

Comment:
@boztopuz I am not sure if I understood your request correctly here.

I am using two environment variable functions but
`get_env_variable` => If there is no environment variable it returns nil (for **ooptional** fields)
`env_has_key` => If there is no environment it returns an error (for **mandatory** fields)

Moreover, these functions do not contain code repetition because I use `get_env_variable` in `env_has_key`.
Also the reason I use two functions is because of the [Single Responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).